src/setup.ts

```diff
@@ -1,21 +1,35 @@
 import 'reflect-metadata';
 import { ApolloServer } from 'apollo-server';
-import { createConnection } from 'typeorm';
+import { Connection, createConnection } from 'typeorm';
 import { typeDefs } from './type-defs';
 import { resolvers } from './resolvers';
 
 export async function startServer() {
   try {
-    const connection = await createConnection();
-    const server = new ApolloServer({
-      typeDefs,
-      resolvers,
-      context: { connection },
-    });
+    const connection = await startDatabase();
+    const server = startApolloServer(connection);
 
     const { url } = await server.listen();
     console.log(`Server ready at ${url}`);
   } catch (e) {
     console.error(e);
   }
 }
+
+async function startDatabase() {
+  return await createConnection({
```

Comment:
```suggestion
function startDatabase() {
  return createConnection({
```

O `createConnection` já retorna uam Promise, então vc pode simplificar. A utilização é a mesma, no fim das contas.

Reply:
mas nesses casos não valeria a pena manter o async pra melhorar legibilidade e facilitar a identificação de que precisa de um await quando for chamar a função, sem precisar ver o retorno das chamadas internas?

Reply:
O Typescript já vai tipar o retorno da função como `Promise<Connection>`. Vai ser a mesma coisa.